src/webhook.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from 'zod'
+import { kycSchemaSchema, kycStatusSchema } from './kyc'
+import { transferStatusResponseSchema } from './transfer'
+
+export enum WebhookEventType {
+  KycStatusEvent = 'WebhookKycStatusEvent',
+  TransferInStatusEvent = 'WebhookTransferInStatusEvent',
+  TransferOutStatusEvent = 'WebhookTransferOutStatusEvent',
+}
+export const webhookEventTypeSchema = z.nativeEnum(WebhookEventType)
+
+export const webhookRequestBodySchema = z.union([
+  z.object({
+    eventType: z.literal(WebhookEventType.KycStatusEvent),
+    provider: z.string(),
+    eventId: z.string(),
+    timestamp: z.string(),
+    address: z.string(),
+    payload: z.object({
+      kycSchema: kycSchemaSchema,
+      kycStatus: kycStatusSchema,
+    }),
+  }),
+  z.object({
+    eventType: z.literal(WebhookEventType.TransferInStatusEvent),
```

Comment:
Does this work to merge the two transfer event schemas?

```suggestion
    eventType: z.union([z.literal(WebhookEventType.TransferInStatusEvent), z.literal(WebhookEventType.TransferOutStatusEvent)])
```

Reply:
Or maybe
```suggestion
    eventType: z.literal(WebhookEventType.TransferInStatusEvent).or(z.literal(WebhookEventType.TransferOutStatusEvent))
```
is a bit more readable?

Reply:
yea good idea!